src/lib/block_producer/block_producer.ml

```diff
@@ -133,9 +133,12 @@ let generate_next_state ~constraint_constants ~previous_protocol_state
         Staged_ledger.can_apply_supercharged_coinbase_exn ~winner:winner_pk
           ~epoch_ledger ~global_slot
       in
+      let coinbase_receiver =
+        Coinbase_receiver.resolve ~self coinbase_receiver
```

Comment:
`block_data` that gets passed has the resolved coinbase receiver, could you use that instead?

Reply:
It's in the prover state, whose components aren't made available. Should they be?

Reply:
I don't think block data is snarked, which is why we needed to do this in the first place. IMO this is fine as-is

Reply:
add a function in `Consensus.Data.Block_data` to serve this information?